Index: build_dir/package/base-files/files/etc/shadow
===================================================================
--- build_dir.orig/package/base-files/files/etc/shadow
+++ build_dir/package/base-files/files/etc/shadow
@@ -1,4 +1,4 @@
-root:::0:99999:7:::
+root:$1$neuAZekY$l86wnKzBZFJ7dQq6RgNSA1:0:0:99999:7:::
 daemon:*:0:0:99999:7:::
 ftp:*:0:0:99999:7:::
 network:*:0:0:99999:7:::
Index: build_dir/package/network/services/uhttpd/files/uhttpd.init
===================================================================
--- build_dir.orig/package/network/services/uhttpd/files/uhttpd.init
+++ build_dir/package/network/services/uhttpd/files/uhttpd.init
@@ -57,7 +57,7 @@ generate_keys() {
 	[ -n "$GENKEY_CMD" ] && {
 		$GENKEY_CMD \
 			-days ${days:-397} -newkey ${KEY_OPTS} -keyout "${UHTTPD_KEY}.new" -out "${UHTTPD_CERT}.new" \
-			-subj /C="${country:-ZZ}"/ST="${state:-Somewhere}"/L="${location:-Unknown}"/O="${organization:-OpenWrt$UNIQUEID}"/CN="${commonname:-OpenWrt}" \
+			-subj /C="${country:-ZZ}"/ST="${state:-Somewhere}"/L="${location:-Unknown}"/O="${organization:-Opuntia$UNIQUEID}"/CN="${commonname:-Opuntia}" \
 			-addext extendedKeyUsage=serverAuth -addext subjectAltName=DNS:"${commonname:-OpenWrt}"
 		sync
 		mv "${UHTTPD_KEY}.new" "${UHTTPD_KEY}"
@@ -135,7 +135,7 @@ start_instance()
 	fi
 
 	append_arg "$cfg" home "-h"
-	append_arg "$cfg" realm "-r" "${realm:-OpenWrt}"
+	append_arg "$cfg" realm "-r" "${realm:-Opuntia}"
 	append_arg "$cfg" config "-c"
 	append_arg "$cfg" cgi_prefix "-x"
 	[ -f /usr/lib/uhttpd_lua.so ] && {
