Index: build_dir/feeds/packages/net/squid/Makefile
===================================================================
--- build_dir.orig/feeds/packages/net/squid/Makefile
+++ build_dir/feeds/packages/net/squid/Makefile
@@ -74,8 +74,11 @@ CONFIGURE_ARGS += \
 	--libexecdir=/usr/lib/squid \
 	--sysconfdir=/etc/squid \
 	--disable-static \
-	--enable-delay-pools \
-	--enable-cache-digests \
+	--enable-storeio=no \
+	--enable-diskio=no \
+	--disable-wccp \
+	--disable-wccpv2 \
+	--disable-htcp \
 	--enable-linux-netfilter \
 	--disable-unlinkd \
 	--enable-x-accelerator-vary \
