Index: build_dir/target/linux/bcm27xx/image/Makefile
===================================================================
--- build_dir.orig/target/linux/bcm27xx/image/Makefile
+++ build_dir/target/linux/bcm27xx/image/Makefile
@@ -224,7 +224,31 @@ define Device/apb3500
 	raspberrypi,4-model-b
   DEVICE_PACKAGES := \
 	cypress-firmware-43455-sdio \
-	cypress-nvram-43455-sdio-rpi-4b \
+	brcmfmac-firmware-43455-sdio-rpi-cm4 \
+	kmod-brcmfmac wpad-basic-wolfssl
+  IMAGE/sysupgrade.img.gz := boot-common | boot-2711 | sdcard-img | gzip | append-metadata
+  IMAGE/factory.img.gz := boot-common | boot-2711 | sdcard-img | gzip
+endef
+ifeq ($(SUBTARGET),bcm2711)
+  TARGET_DEVICES += apb3500
+  TARGET_DEVICES += apb3350
+endif
+
+define Device/apb3350
+  DEVICE_VENDOR := ImageStream
+  DEVICE_MODEL := apb3350
+  DEVICE_VARIANT := (64bit)
+  KERNEL_IMG := kernel8.img
+  DEVICE_DTS := broadcom/bcm2711-is-apb3350
+  DEVICE_RENAME_DTS := bcm2711-rpi-cm4
+  SUPPORTED_DEVICES := \
+	imagestream,apb3350 \
+	imagestream,apb3500 \
+ 	raspberrypi,4-compute-module \
+ 	raspberrypi,4-model-b
+   DEVICE_PACKAGES := \
+   	cypress-firmware-43455-sdio \
+	brcmfmac-firmware-43455-sdio-rpi-cm4 \
 	kmod-brcmfmac wpad-basic-wolfssl
   IMAGE/sysupgrade.img.gz := boot-common | boot-2711 | sdcard-img | gzip | append-metadata
   IMAGE/factory.img.gz := boot-common | boot-2711 | sdcard-img | gzip
