--- a/package/base-files/Makefile
+++ b/package/base-files/Makefile
@@ -46,7 +46,7 @@ define Package/base-files
 	+SELINUX:busybox-selinux +!SELINUX:busybox
   TITLE:=Base filesystem for OpenWrt
   URL:=http://openwrt.org/
-  VERSION:=$(PKG_RELEASE)~$(lastword $(subst -, ,$(REVISION)))
+  VERSION:=$(PKG_RELEASE)~$(subst r,,${lastword $(subst -, ,$(REVISION))})
 endef

 define Package/base-files/conffiles
@@ -256,7 +256,7 @@ ifneq ($(CONFIG_USE_APK),)

 	rm -f $(1)/etc/uci-defaults/13_fix-group-user
 	rm -f $(1)/sbin/pkg_check
-	echo $(PKG_RELEASE)~$(lastword $(subst -, ,$(REVISION))) >$(TMP_DIR)/base-files.version
+	echo $(PKG_RELEASE)~$(subst r,,${lastword $(subst -, ,$(REVISION))}) >$(TMP_DIR)/base-files.version
 else
 	$(if $(CONFIG_CLEAN_IPKG),, \
 		mkdir -p $(1)/etc/opkg; \
