Index: build_dir/package/network/services/ppp/Makefile
===================================================================
--- build_dir.orig/package/network/services/ppp/Makefile
+++ build_dir/package/network/services/ppp/Makefile
@@ -16,7 +16,7 @@ PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/ppp-project/ppp
 PKG_SOURCE_DATE:=2024-12-31
 PKG_SOURCE_VERSION:=9f612dc02c34509f062ed63b60bcc7e937e25178
-PKG_MIRROR_HASH:=677b71d23b668db986146e13b0c651f2ac506eb4fb244ffba1ff406cbae3511b
+PKG_MIRROR_HASH:=381668ce1547afeaf599f08ae37935bfd3d5841747c62bca5a870c11e3e9bedb
 
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
 PKG_LICENSE:=BSD-4-Clause
